spyder/plugins/ipythonconsole/widgets/tests/test_kernelconnect.py

```diff
@@ -0,0 +1,201 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright © Spyder Project Contributors
+#
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+# -----------------------------------------------------------------------------
+
+"""Tests for the Existing Kernel Connection widget."""
+
+# Third party imports
+import sys
+import os
+import pytest
+from qtpy.QtCore import Qt
+from qtpy.QtWidgets import QDialogButtonBox
+import time
+
+# Local imports
+from spyder.plugins.ipythonconsole.widgets.kernelconnect import (
+    KernelConnectionDialog
+)
+from spyder.config.main import CONF
+
+
+# =============================================================================
+# Fixtures
+# =============================================================================
+@pytest.fixture
+def connection_dialog_factory(qtbot, request):
+    """Set up kernel connection dialog."""
+    class DialogFactory(object):
+        def get_default_dialog(self):
+            dialog = KernelConnectionDialog()
+            request.addfinalizer(dialog.close)
+            return dialog
+
+        def submit_filled_dialog(self, use_keyfile, save_settings):
+            dlg = self.get_default_dialog()
+
+            # fill out cf path
+            dlg.cf.clear()
+            qtbot.keyClicks(dlg.cf, cf_path)
+            # check the remote kernel box
+            dlg.rm_group.setChecked(True)
+            # fill out hostname
+            dlg.hn.clear()
+            qtbot.keyClicks(dlg.hn, hn)
+            # fill out port
+            dlg.pn.clear()
+            qtbot.keyClicks(dlg.pn, str(pn))
+            # fill out username
+            dlg.un.clear()
+            qtbot.keyClicks(dlg.un, un)
+
+            if use_keyfile:
+                # select ssh keyfile radio
+                dlg.kf_radio.setChecked(True)
+                assert dlg.kf.isEnabled()
+
+                # fill out ssh keyfile
+                dlg.kf.clear()
+                qtbot.keyClicks(dlg.kf, kf)
+
+                # fill out passphrase
+                dlg.kfp.clear()
+                qtbot.keyClicks(dlg.kfp, kfp)
+            else:
+                # select password radio
+                dlg.pw_radio.setChecked(True)
+                assert dlg.pw.isEnabled()
+
+                # fill out password
+                dlg.pw.clear()
+                qtbot.keyClicks(dlg.pw, pw)
+
+            # check save connection settings
+            dlg.save_layout.setChecked(save_settings)
+
+            # Press ok and save connection settings
+            qtbot.mouseClick(dlg.accept_btns.button(QDialogButtonBox.Ok),
```

Comment:
This one should not be part of the fixture IMO, and thus I specifically did not include it in my previous request. Conceptually, the fixture is the pre-filled dialog object, not the "action" of dialog submission. Another test function (maybe even one you'd want to write here) might want to hit the cancel button instead on the dialog, toggle some of the fields, close it via the close button, or do something else.